src/lib.rs

```diff
@@ -84,10 +91,18 @@ macro_rules! base_main {
                             .filter(|day| days.contains(&format!("day{}", day).as_str()))
                             .collect()
                     } else {
-                        parse!(extract_day {}; $( $tail )*)
-                            .iter()
-                            .map(|s| &s[3..])
-                            .collect()
+                        if opt.is_present("all") {
```

Comment:
this could probably just be an else if and an else, I don't think it needs to be nested